core/node/libp2p/rcmgr.go

```diff
@@ -1,40 +1,53 @@
 package libp2p
 
 import (
+	"context"
 	"errors"
 	"fmt"
+	"github.com/ipfs/go-ipfs/repo"
+	"github.com/libp2p/go-libp2p-core/network"
+	"go.uber.org/fx"
 	"os"
+	"path/filepath"
 
 	"github.com/libp2p/go-libp2p"
 	rcmgr "github.com/libp2p/go-libp2p-resource-manager"
 )
 
-func ResourceManager() func() (Libp2pOpts, error) {
-	return func() (opts Libp2pOpts, err error) {
+func ResourceManager() func(fx.Lifecycle, repo.Repo) (network.ResourceManager, Libp2pOpts, error) {
+	return func(lc fx.Lifecycle, repo repo.Repo) (network.ResourceManager, Libp2pOpts, error) {
 		var limiter *rcmgr.BasicLimiter
+		var opts Libp2pOpts
 
-		limitsIn, err := os.Open("./limits.json")
+		// FIXME(BLOCKING): Get repo path.
```

Comment:
Question: Is this supposed to be the `.ipfs/` directory?

Reply:
No, all this is wrong, but this is just a draft and the feedback is mostly to see if the command is working.